### PR TITLE
Added Rsyslog and Syslog-ng.

### PR DESCRIPTION
**Pull request by asenkowski**
Adds support for Rsyslog and Syslog-ng.

### CODE REVIEW

File: plugins/rsyslog.yaml [hunk 1]
```diff
@@ -0,0 +1,105 @@
+# Plugin Info
+version: 0.0.1
+title: Rsyslog
+description: Log parser for Rsyslog
+parameters:
+  - name: listen_address
+    label: Listen Address
+    description: An rsyslog address of the form `<ip>:<port>`
```
Comment (jsirianni) [inline plugins/rsyslog.yaml:8]: nitpick
```suggestion
    description: A rsyslog address of the form `<ip>:<port>`
```

File: plugins/rsyslog.yaml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  # structured_data key=value pairs or message fields contain
+  # a \n. This is unlikly, but not impossible. It has been
+  # observed with rfc5424 structured_data field.
+  # '<' can be relied on to be the first character of every
+  # syslog message for both rfc 5424 and 3164
+  - id: handle_new_lines
+    type: recombine
+    combine_field: $record
+    is_first_entry: "$record startsWith '<'"
+    output: return_router
```
Comment (jsirianni) [inline plugins/rsyslog.yaml:75]: The original Syslog implementation needs to handle special situations where syslog entries are multi line. This should not be the case for a dedicated rsyslog plugin. 

```suggestion
  - id: rsyslog_input
    type: tcp_input
    listen_address: {{ $listen_address }}
    labels:
      log_type: rsyslog
      plugin_id: {{ .id }}
    output: return_router
```

File: plugins/rsyslog.yaml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    regex: '(?P<message1>.*)(?P<return>\\r)(?P<message2>.*)'
+    output: return_restructurer
+
+  - id: return_restructurer
+    type: restructure
+    ops:
+      - add:
+          field: $record
+          value_expr: '$record.message1 + "\\\\r" + $record.message2'
+    output: rsyslog_parser
```
Comment (jsirianni) [inline plugins/rsyslog.yaml:97]: The original Syslog implementation needs to handle special situations where return characters were present in the log entry. I do not believe we will run into this issue when targeting rsyslog specifically. This special handling was specifically for NetMotion over syslog. https://github.com/observIQ/stanza-plugins/pull/244

```suggestion

```

File: plugins/rsyslog.yaml (hunk 2 above)
Comment (jsirianni) [reply plugins/rsyslog.yaml:75]: rsyslog_input will need to output to `rsyslog_parser`, see next comment below V V V

**Review by jsirianni — CHANGES_REQUESTED** [review]
I made some suggestions to rsyslog, but they also apply to syslog-ng.

**Review by jsirianni — APPROVED** [review]

File: plugins/rsyslog.yaml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+# Set Defaults
+# {{$listen_address := default "0.0.0.0:514" .listen_address}}
+# {{$connection_type := default "udp" .connection_type}}
+# {{$protocol := default "rfc5424 (IETF)" .protocol}}
+# {{$location := default "UTC" .location}}
+
+# Pipeline Template
+pipeline:
+# {{ if eq $connection_type "udp" }}
+  - id: rsyslog_input
```
Comment (ericwholt) [inline plugins/rsyslog.yaml:46]: Id's should be unique.
```suggestion
  - id: rsyslog_udp_input
```

File: plugins/rsyslog.yaml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    type: udp_input
+    listen_address: {{ $listen_address }}
+    labels:
+      log_type: rsyslog
+      plugin_id: {{ .id }}
+    output: rsyslog_parser
+# {{ end }}
+
+# {{ if eq $connection_type "tcp" }}
+  - id: rsyslog_input
```
Comment (ericwholt) [inline plugins/rsyslog.yaml:56]: Id's should be unique.
```suggestion
  - id: rsyslog_tcp_input
```

File: plugins/syslogng.yaml [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+# Set Defaults
+# {{$listen_address := default "0.0.0.0:514" .listen_address}}
+# {{$connection_type := default "udp" .connection_type}}
+# {{$protocol := default "rfc5424 (IETF)" .protocol}}
+# {{$location := default "UTC" .location}}
+
+# Pipeline Template
+pipeline:
+# {{ if eq $connection_type "udp" }}
+  - id: syslogng_input
```
Comment (ericwholt) [inline plugins/syslogng.yaml:46]: Id's should be unique.
```suggestion
  - id: syslogng_udp_input
```

File: plugins/syslogng.yaml [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    type: udp_input
+    listen_address: {{ $listen_address }}
+    labels:
+      log_type: syslogng
+      plugin_id: {{ .id }}
+    output: syslogng_parser
+# {{ end }}
+
+# {{ if eq $connection_type "tcp" }}
+  - id: syslogng_input
```
Comment (ericwholt) [inline plugins/syslogng.yaml:56]: Id's should be unique.
```suggestion
  - id: syslogng_tcp_input
```

**Review by ericwholt — CHANGES_REQUESTED** [review]
Looks good just a some minor changes requested.

**Review by ericwholt — COMMENTED** [review]
👍

**Review by ericwholt — COMMENTED** [review]
LGTM

**Pull request merged**